pkg/lib/cockpit-upload-helper.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+
+function debug(...args: unknown[]) {
+    if (window.debugging == 'all' || window.debugging?.includes('upload'))
+        console.debug('debug[upload]', ...args);
```

Comment:
These 2 added lines are not executed by any test.

Reply:
Normally this is:
```
pkg/systemd/services.jsx:        console.debug.apply(console, arguments);
```

Can we, keep things consistent :)

Reply:
We also have:
```
pkg/lib/cockpit.js:            spawn_debug("process closed:", JSON.stringify(options));
pkg/lib/cockpit.js:                spawn_debug("process output:", data);
```

In a browser you filter log levels by clicking what you want. so what is console.debug is quite obvious to see.

Reply:
Ugh, but firewall page does:

```
    if (window.debugging == "all" || window.debugging?.includes("firewalld")) // not-covered: debugging
        console.debug("firewalld:", ...arguments); // not-covered: debugging
}
```

Which seems good

Reply:
Right, please drop the redundant "debug[]" part of the string.

Reply:
`.apply()` and `arguments` are both considered gauche by tsc